### PR TITLE
Bug fix: Admin changes are not reflecting in proxy

### LINKED ISSUE #1509: runtime changes not being applied
### Summary

Very odd one, I have Kong running in a AWS autoscaling group using docker container. At the moment, I only have a single instances in the cluster. I can make changes via the API, I can call the Admin API and see the changes have been applied. But when I call actual proxy, the configuration does looks like it's been applied. Example, creating and API, view its config and everything looks fine, by call the API and it throws a API not found. Running kong reload does not work, but if i kong restart or simply kill the container, the configuration has been applied.
### Steps To Reproduce
1. Simply make a change to API and check it has taken place
### Additional Details & Logs
- Kong version: 0.8.3
- Operating System: CoreOS 
- Kong configuration 

``` YAML
#
# Kong Configuration
#
proxy_listen: "0.0.0.0:8000"
proxy_listen_ssl: "0.0.0.0:8443"
admin_api_port: "0.0.0.0:8001"
send_anonymous_reports: false
memory_cache_size: 128
cluster_listen: "0.0.0.0:7946"
cluster_listen_rpc: "127.0.0.1:7373"
cluster:
  ttl_on_failure: 120
database: postgres
postgres:
  host: "HOSTNAME"
  port: 5432
  database: DBNAME
  user: USERNAME
  password: PASSWORD
ssl_cert_path: /etc/secrets/proxy.pem
ssl_key_path: /etc/secrets/proxy-key.pem
nginx: |
  {{user}}
  worker_processes auto;
  error_log /dev/stderr error;
  daemon off;
  worker_rlimit_nofile {{auto_worker_rlimit_nofile}};
  env KONG_CONF;
  env PATH;
  events {
    worker_connections {{auto_worker_connections}};
    multi_accept on;
  }
  http {
    resolver {{dns_resolver}} ipv6=off;
    charset UTF-8;

    include /etc/nginx/conf/logging-formats.conf;
    include /etc/nginx/conf/logging.conf;

    # Timeouts
    keepalive_timeout 60s;
    client_header_timeout 60s;
    client_body_timeout 60s;
    send_timeout 60s;

    # Proxy settings
    proxy_buffer_size 128k;
    proxy_buffers 4 256k;
    proxy_busy_buffers_size 256k;
    proxy_ssl_server_name on;

    # IP Address
    real_ip_recursive on;

    # Other Settings
    client_max_body_size 0;
    underscores_in_headers on;
    reset_timedout_connection on;
    tcp_nopush on;

    ################################################
    #  The following code is required to run Kong  #
    # Please be careful if you'd like to change it #
    ################################################
    # Lua Settings
    lua_package_path ';;';
    lua_code_cache on;
    lua_max_running_timers 4096;
    lua_max_pending_timers 16384;
    lua_shared_dict reports_locks 100k;
    lua_shared_dict cluster_locks 100k;
    lua_shared_dict cluster_autojoin_locks 100k;
    lua_shared_dict cache {{memory_cache_size}}m;
    lua_shared_dict cassandra 1m;
    lua_shared_dict cassandra_prepared 5m;
    lua_socket_log_errors off;
    {{lua_ssl_trusted_certificate}}
    init_by_lua_block {
      kong = require "kong"
      kong.init()
    }
    init_worker_by_lua_block {
      kong.init_worker()
    }
    server {
      server_name _;
      listen {{proxy_listen}} proxy_protocol;
      listen {{proxy_listen_ssl}} ssl proxy_protocol;
      real_ip_header proxy_protocol;
      set_real_ip_from 10.0.0.0/8;
      real_ip_recursive on;
      ssl_certificate_by_lua_block {
        kong.ssl_certificate()
      }
      ssl_certificate {{ssl_cert}};
      ssl_certificate_key {{ssl_key}};
      ssl_protocols TLSv1.1 TLSv1.2;
      location / {
        default_type 'text/plain';
        # These properties will be used later by proxy_pass
        set $upstream_host nil;
        set $upstream_url nil;
        # Authenticate the user and load the API info
        access_by_lua_block {
          kong.access()
        }
        proxy_set_header X-Real-IP $proxy_protocol_addr;
        proxy_set_header X-Forwarded-For $proxy_protocol_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $upstream_host;
        proxy_pass $upstream_url;
        proxy_pass_header Server;
        # Add additional response headers
        header_filter_by_lua_block {
          kong.header_filter()
        }
        # Change the response body
        body_filter_by_lua_block {
          kong.body_filter()
        }
        # Log the request
        log_by_lua_block {
          kong.log()
        }
      }
      location /robots.txt {
        return 200 'User-agent: *\nDisallow: /';
      }
      error_page 500 502 503 504 /50x;
      location = /50x {
        internal;
        content_by_lua_block {
          require("kong.core.error_handlers")(ngx)
        }
      }
    }
    server {
      listen {{admin_api_listen}} ssl proxy_protocol;
      real_ip_header proxy_protocol;
      set_real_ip_from 10.0.0.0/8;
      client_max_body_size 10m;
      client_body_buffer_size 10m;
      ssl_certificate {{ssl_cert}};
      ssl_certificate_key {{ssl_key}};
      ssl_protocols TLSv1.1 TLSv1.2;
      location / {
        include /etc/nginx/conf/auth.conf;
        default_type application/json;
        content_by_lua_block {
          ngx.header["Access-Control-Allow-Origin"] = "*"
          if ngx.req.get_method() == "OPTIONS" then
            ngx.header["Access-Control-Allow-Methods"] = "GET,HEAD,PUT,PATCH,POST,DELETE"
            ngx.header["Access-Control-Allow-Headers"] = "Content-Type"
            ngx.exit(204)
          end
          local lapis = require "lapis"
          lapis.serve "kong.api.app"
        }
      }
      location /nginx_status {
        internal;
        access_log off;
        stub_status;
      }
      location /robots.txt {
        return 200 'User-agent: *\nDisallow: /';
      }
    }
  }
```

### PR DESCRIPTION
**Pull request by dheerajdwivedi**
### Summary

In case of admin ip changed from default.
The admin api changes are not being pushed to serf.
As the default ip (127.0.0.1) is hardcoded in the script.

### Full changelog

* Set the new admin ip if changed from default.

### Issues resolved

Fix #1509 

Setting the admin IP if changed from default.
Fix for issue: https://github.com/Mashape/kong/issues/1509

### CODE REVIEW

**Comment by thefosk** [comment]
@dheerajdwivedi can you rebase on the `next` branch?

**Comment by dheerajdwivedi** [comment]
Done !
Message while creating pull request is confusing.
`**Note: new features are to be opened against next, hotfixes against master.**`

File: kong/cmd/utils/prefix_handler.lua [hunk 1]
```diff
@@ -223,7 +223,13 @@ local function prepare_prefix(kong_config, nginx_custom_template_path)
   if not resty_bin then return nil, err end
 
   log.verbose("saving serf shell script handler to %s", kong_config.serf_event)
-  local script = fmt(script_template, "127.0.0.1", kong_config.admin_port, resty_bin)
+  -- setting serf admin ip
+  local admin_ip = kong_config.admin_ip
+  if kong_config.admin_ip == '0.0.0.0' then
```
Comment (thefosk) [inline kong/cmd/utils/prefix_handler.lua]: Doesn't it work with `0.0.0.0` without converting it to `127.0.0.1` ?

**Comment by thefosk** [comment]
One last thing: can you use double quotes `"` instead of single quotes `'` ? @dheerajdwivedi

**Comment by dheerajdwivedi** [comment]
Done.
But build failed, is this usual failure ?
`[  FAILED  ] 1 test, listed below:`
`[  FAILED  ] spec/02-integration/01-cmd/02-start_stop_spec.lua @ 156: kong start/stop dnsmasq recovers from expired dnsmasq.pid file`

**Comment by dheerajdwivedi** [comment]
@thefosk Build failure is inconsistent, there has not been any change in logic only cosmetics changes are done after last successful build.
Build environment is not stable.
> Please re-run the build.

**Comment by thefosk** [comment]
@dheerajdwivedi it's okay - those errors are related to something else.

**Pull request merged**